homeassistant/components/config/entity_registry.py

```diff
@@ -223,8 +223,14 @@ def websocket_remove_entity(hass, connection, msg):
 
 
 @callback
-def _entry_dict(entry):
+def _entry_dict(entry: er.RegistryEntry):
     """Convert entry to API format."""
+    # If entity name we always want to have a name.
+    if entry.has_entity_name:
+        name = entry.name or entry.original_name
+    else:
+        name = entry.name
```

Comment:
Instead of this, core should send `entry.original_name` to frontend, frontend can then use whatever logic is needed to render a complete name.